### PR TITLE
Correct parenthetical type spacing

### LINKED ISSUE #215: [OCaml] Odd spaces in parenthesized types
As of 6f71f36:

### Snippet 1

The following
```ocaml
type query = (string * string list) list
```
gets formatted as
```ocaml
type query = (string * string list ) list
```
(This also happens without `string *`, even if I would probably then not use parentheses.)
(This also happens without the first `list`.)

### Snippet 2

The following
```ocaml
type x = ('any Slug.t -> bool) -> float
```
gets formatted as
```ocaml
type x = ( 'any Slug.t -> bool) -> float
```
(This does not happen without `'any`.)

### PR DESCRIPTION
**Pull request by Xophmeister**
Resolves #215

### CODE REVIEW

**Comment by Xophmeister** [comment]
Yikes... Let me fix the merge conflicts

File: languages/ocaml.scm [hunk 1]
```diff
@@ -350,13 +350,16 @@
...
+  (_)
   .
   (type_variable) @prepend_space
 )
+(
+  (type_variable)
+  .
+  "." @append_space
+)
+
```
Comment (nbacquey) [inline languages/ocaml.scm:362]: I do really understand the reason for this modification, and the one above. Could you clarify?

Comment (Xophmeister) [reply languages/ocaml.scm:362]: (Disclaimer: I don't know the OCaml terminology, so I'll have to spell things out verbosely :sweat_smile:)

The original rule:
```scm
(
  [
    "("
    "["
  ]* @do_nothing
  .
  (type_variable) @prepend_space
)
```
..._should_ have been enough to address the spacing in the `type t: ('x -> x)`-case. However, it clearly wasn't working. As such, I changed it to this:

```scm
(
  (_)
  .
  (type_variable) @prepend_space
)
```

That worked, but it broke the spacing of type variables followed by a full stop, which is an anonymous node. i.e.:
```ocaml
(* This... *)
let my_const : 'a 'b. 'a -> 'b -> 'a = Fun.const

(* Got formatted as... *)
let my_const : 'a 'b.'a -> 'b -> 'a = Fun.const
```

I tried changing the first half of the tuple pattern to `[(_) "."]`, but that didn't work either. As such, I had to write an explicit rule for it to target type variables followed by a full stop, hence:
```scm
(
  (type_variable)
  .
  "." @append_space
)
```

There may be a better solution, but this worked.

Comment (nbacquey) [reply languages/ocaml.scm:362]: I think that, for the sake of the maintainability of the query file, it's better to remove from existing rules than add new rules, and I think the problem can be addressed by the former. Let me draft something

Comment (nbacquey) [reply languages/ocaml.scm:362]: My attempt is at #258

Comment (Xophmeister) [reply languages/ocaml.scm:362]: That looks great :100: I'll close this PR

**Comment by Xophmeister** [comment]
Subsumed by #258

**Pull request closed without merging**